cpp/demo/mixed_poisson/main.cpp

```diff
@@ -221,34 +260,51 @@ int main(int argc, char* argv[])
         = fem::locate_dofs_topological(
             *mesh->topology(), {*V0->dofmap(), *W0->dofmap()}, 1, nfacets);
 
-    // Create boundary condition for \sigma. \sigma \cdot n will be
-    // constrained to to be equal to the normal component of g. The
+    // Create boundary condition for $\sigma. $\sigma \cdot n$ will be
+    // constrained to to be equal to the normal component of $g$. The
     // boundary conditions are applied to degrees-of-freedom ndofs, and
-    // V0 is the subspace that is constrained.
+    // `V0` is the subspace that is constrained.
     fem::DirichletBC<T> bc(g, ndofs, V0);
 
-    // Create integration domain data for u0 boundary condition (applied
-    // on the ds(1) in the UFL file). First we get facet data
+    // Create integration domain data for `u0` boundary condition
+    // (applied on the `ds(1)` in the UFL file). First we get facet data
     // integration data for facets in dfacets.
     std::vector<std::int32_t> domains = fem::compute_integration_domains(
         fem::IntegralType::exterior_facet, *mesh->topology(), dfacets);
 
-    // Create data structure for the ds(1) integration domain in form
+    // Create data structure for the `ds(1)` integration domain in form
     // (see the UFL file). It is for en exterior facet integral (the key
     // in the map), and exterior facet domain marked as '1' in the UFL
-    // file, and 'domains' holds the necessary data to perform
+    // file, and `domains` holds the necessary data to perform
     // integration of selected facets.
     std::map<
         fem::IntegralType,
         std::vector<std::pair<std::int32_t, std::span<const std::int32_t>>>>
         subdomain_data{{fem::IntegralType::exterior_facet, {{1, domains}}}};
 
+    // Since we are doing a `ds(1)` integral on mesh and `u0` is defined
+    // on the `submesh`, we must provide an "entity map" relating cells
+    // in `submesh` to entities in `mesh`. This is simply the "inverse"
+    // of `submesh_to_mesh`:
+    auto facet_imap = mesh->topology()->index_map(fdim);
+    assert(facet_imap);
+    std::size_t num_facets = mesh->topology()->index_map(fdim)->size_local()
+                             + mesh->topology()->index_map(fdim)->num_ghosts();
+    std::vector<std::int32_t> mesh_to_submesh(num_facets, -1);
```

Comment:
Maybe add a note regarding why we use -1 as default here (until we use a revamped `EntityMap`-class).